lib/src/Uncertainty/Algorithm/Simulation/EventSimulation.cxx

```diff
@@ -74,6 +76,24 @@ EventSimulation * EventSimulation::clone() const
 /*  Event accessor */
 RandomVector EventSimulation::getEvent() const
 {
+  // For an Intersection/UnionEvent, we get its composedEvent if it can be built.
+  // This allows access to the level function for derived classes which use them (SubsetSampling for example).
+  // For other events, and Intersection/UnionEvents which cannot be described as a composedEvent,
+  // we default to returning the event_ attribute.
+  if (event_.getImplementation()->getClassName() == "IntersectionEvent")
+  {
+    IntersectionEvent *intersectionEvent = static_cast<IntersectionEvent*>(event_.getImplementation().get());
+    const Bool hasComposedEvent = intersectionEvent->buildComposedEvent();
+    if (hasComposedEvent)
+      return intersectionEvent->getComposedEvent();
+  }
+  if (event_.getImplementation()->getClassName() == "UnionEvent")
+  {
+    UnionEvent *unionEvent = static_cast<UnionEvent*>(event_.getImplementation().get());
+    const Bool hasComposedEvent = unionEvent->buildComposedEvent();
+    if (hasComposedEvent)
+      return unionEvent->getComposedEvent();
+  }
```

Comment:
these can be avoid if getComposedEvent is a virtual method of RandomvectorImplementation
(to be fair it would also yield a cleaner code in the IntersectionEvent/UnionEvent of master branch)

Reply:
Done. You were right, this simplifies the code.